src/App.tsx

```diff
@@ -16,35 +17,95 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType | ''>('');
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  enum SortType {
+    ALPHABET = 'alphabet',
+    LENGTH = 'length',
+    NONE = 'none',
+  }
+
+  const visibleGoods = [...goodsFromServer].sort(
+    (good1: string, good2: string) => {
+      switch (sortField) {
+        case SortType.ALPHABET:
+          return good1.localeCompare(good2);
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    },
+  );
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  const reverse = () => {
+    setIsReversed(prevState => !prevState);
+  };
+
+  const reset = () => {
+    setSortField('');
+    setIsReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={classNames('button is-info', {
+            'is-light': sortField !== SortType.ALPHABET,
+          })}
+          onClick={() => {
+            setSortField(SortType.ALPHABET);
+          }}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={classNames('button is-success', {
+            'is-light': sortField !== SortType.LENGTH,
+          })}
+          onClick={() => {
+            setSortField(SortType.LENGTH);
+          }}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={reverse}
```

Comment:
For the 'Reverse' button, make sure that the active state is visually distinct as well to improve user experience. [UX]